app/main.py

```diff
@@ -1,3 +1,20 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable, Any
+
+
+def cache(func: Callable) -> Callable:
+    storage = {}
+
+    def inner(*args) -> Any:
+
+        func_arguments_and_results = storage.get(func)
+        if func in storage and args in func_arguments_and_results:
```

Comment:
I think that this check: `func in storage` is unused, you can just remove it

Reply:
instead of `args in func_arguments_and_results` you can just use `args in storage`